pkg/didcomm/protocol/didexchange/states.go

```diff
@@ -559,6 +555,99 @@ func (ctx *context) getDIDDocAndConnection(pubDID string, routerConnections []st
 	return docResolution.DIDDocument, connection, nil
 }
 
+func createNewKeyAndVerificationMethod(didDoc *did.Doc, keyType kms.KeyType, keyManager kms.KeyManager) error {
+	vmType := getVerMethodType(keyType)
+
+	kid, pubKeyBytes, err := keyManager.CreateAndExportPubKeyBytes(keyType)
+	if err != nil {
+		return err
+	}
+
+	pubKeyBytes, err = convertPubKeyBytes(pubKeyBytes, keyType)
+	if err != nil {
+		return err
+	}
+
+	vm := did.VerificationMethod{
+		ID:    "#" + kid,
+		Type:  vmType,
+		Value: pubKeyBytes,
+	}
+
+	didDoc.VerificationMethod = append(didDoc.VerificationMethod, vm)
+
+	// TODO replace below authentication with a KeyAgreement method when/if DID attachment doesn't require to be signed
+	//  as per https://github.com/hyperledger/aries-rfcs/pull/626. If the PR is declined, then only remove this comment.
+	//  KeyAgreement is needed for envelope encryption regardless. It must be added in a future change.
+	didDoc.Authentication = append(didDoc.Authentication, *did.NewReferencedVerification(&vm, did.Authentication))
+
+	return nil
+}
+
+/* func extractKeyTypeFromOpts(createDIDOpts *vdrapi.DIDMethodOpts) (kms.KeyType, error) {
+	var keyType kms.KeyType
+
+	k := createDIDOpts.Values[keyvdr.KeyType]
+	if k != nil {
+		var ok bool
+		keyType, ok = k.(kms.KeyType)
+
+		if !ok {
+			return "", errors.New("keyType is not kms.KeyType")
+		}
+
+		return keyType, nil
+	}
+
+	return "", errors.New("keyType option is needed for empty didDoc.VerificationMethod")
+} */
```

Comment:
a follow up pr will be created to add key vdr tests for keys other than Ed25519 and add default keys creation in Aries by specifying the key type where this code will be uncommented/used.